### PR TITLE
bug 1246090 - [System] Remove bookmarks and their pin on Home correct…

### PR DESCRIPTION
**Pull request by rexboy7**
…ly even when AppDeck is not executing.

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ee3b76ad5b4a84ef50c9cc67fa12ceecc70b008f

File: tv_apps/smart-system/js/bookmark_handler.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/* global BookmarkManager */
+
+(function(exports) {
+  'use strict';
+
+var BookmarkHandler = {
```
Comment (luke-chang) [inline tv_apps/smart-system/js/bookmark_handler.js]: Miss an indent.

File: tv_apps/app-deck/js/context_menu.js [hunk 2]
```diff
@@ -120,11 +120,11 @@
...
         });
-      } else if (type === 'bookmark') {
-        this.sendMessage('unpin', {
-          url: cardElem.dataset.url
-        });
       }
+      // Unpinning bookmarks are sended in smart-system/bookmark_handler.js
+      // when datastore.onremove is fired. So  we don't need to send unpin
+      // for that now.
+
```
Comment (luke-chang) [inline tv_apps/app-deck/js/context_menu.js]: nit: remove this blank line.

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b1b6715caed3606336de852f04bb028e8aa1e219

**Pull request merged**